File: pset6/readability/readability.py
from cs50 import get_string


def main():  # MAIN function
    s = get_string("Text : ")
    letters = count_letters(s)  # calls the function to count letters
    words = count_words(s)  # calls the function to count words
    sentences = count_sentences(s)  # Calls the function to count sentences
    w = 100/words
    L = letters * w
    S = sentences / words
    S *= 100
    I = 0.0588 * L - 0.296 * S - 15.8  # The Coleman-Liau index formula
    I = round(I)
    if I < 1:  # Print segment
        print("Before Grade 1")
    elif I >= 1 and I <= 16:
        print(f"Grade {I}")
    else:
        print("Grade 16+")


def count_letters(s):   # Function to count letters
    letters = 0
    for i in s:
        if 'a' <= i and i <= 'z':
            letters += 1
        if 'A' <= i and i <= 'Z':
            letters += 1
    return letters
    

def count_words(s):  # function to count words
    # words = 1
    # for i in s:
    #     if s[i] == ' '
    #         words += 1

    return len(s.split())


def count_sentences(s):  # function to count words
    sentences = 0
    for i in s:
        if i == '.' or i == '!' or i == '?':
            sentences += 1
    return sentences
    

main()

File: pset2/readability/readability.c
#include<stdio.h>
#include<cs50.h>
#include <string.h>
#include<math.h>

int count_letters(string s);                            //function declereations
int count_words(string s);
int count_sentences(string s);

int main(void)
{
    string s = get_string("Text : ");
    int letters = count_letters(s);
    int words = count_words(s);
    int sentences = count_sentences(s);
    float w = (float) 100 / words;
    float L = letters * w;
    float S = (float) sentences / words;
    S *= 100;
    float I = 0.0588 * L - 0.296 * S - 15.8;            //The Coleman-Liau index formula
    int index = round(I);

    //printf ("%i \n %i \n %i \n %i",letters,words,sentences,index);
    if (index < 1)                                      //printing segmet
    {
        printf("Before Grade 1");                       //for grade being less than 1
    }
    else if (index >= 1 && index <= 16)                 //for grade being in between 1 and 16
    {
        printf("Grade %i", index);
    }
    else                                                //for grade over 16
    {
        printf("Grade 16+");
    }
    printf("\n");
}

int count_letters(string s)                             //count the number of letters
{
    int letters = 0;
    for (int i = 0; i < strlen(s); i++)
    {
        if ('a' <= s[i] && s[i] <= 'z')
        {
            letters++;
        }
        if ('A' <= s[i] && s[i] <= 'Z')
        {
            letters++;
        }
    }
    return letters;
}

int count_words(string s)                               //count the number of words
{
    int words = 1;
    for (int i = 0; i < strlen(s); i++)
    {
        if (s[i] == ' ')
        {
            words++;
        }
    }
    return words;
}

int count_sentences(string s)                           //count the number of sentences
{
    int sentences = 0;
    for (int i = 0; i < strlen(s); i++)
    {
        if (s[i] == '.' || s[i] == '!' || s[i] == '?')
        {
            sentences++;
        }
    }
    return sentences;
}